src/common/libutil/fluid.c

```diff
@@ -97,6 +146,114 @@ int fluid_generate (struct fluid_generator *gen, fluid_t *fluid)
     return 0;
 }
 
+/*  F58 encoding.
+ */
+/*  Compute base58 encoding of id in *reverse*
+ *  Return number of characters written into buf
+ */
+static int b58revenc (char *buf, int bufsz, fluid_t id)
+{
+    int index = 0;
+    memset (buf, 0, bufsz);
+    if (id == 0) {
+        buf[0] = b58digits[0]; return 1;
+    }
+    while (id > 0) {
+        int rem = id % 58;
+        buf[index++] = b58digits[rem];
+        id = id / 58;
+    }
+    return index;
+}
+
+static int fluid_f58_encode (char *buf, int bufsz, fluid_t id)
+{
+    int count;
+    char b58reverse[13];
+    int index = 0;
+    if (buf == NULL || bufsz <= 0) {
+        errno = EINVAL;
+        return -1;
+    }
+    if (bufsz <= strlen (f58_prefix) + 1) {
+        errno = ENOSPC;
```

Comment:
I've used ENOSPC (No space left on device) in this case, but then somebody came up with EOVERFLOW (Value too large for defined data type) in some of our code and I thought that was more appropriate?  Either is fine but I thought I'd throw that out there.

Reply:
Yeah, I think I like EOVERFLOW as well. Thanks!

Reply:
Sorry to jump in without having to review the entire PR. 

I have been using `ERANGE` as well and I think `ERANGE` or `EDOM` could be another choice.

https://en.wikipedia.org/wiki/Errno.h

Reply:
Well, now that I think about it, in this particular case, `EOVERFLOW` would be best since this is a buffer being passed, not a number being passed. Sorry for the noise.

Reply:
No problem, thanks for the feedback!